lib/Language/Haskell/Stylish/Step/Imports.hs

```diff
@@ -103,6 +119,27 @@ data LongListAlign
     | Multiline -- multiline
     deriving (Eq, Show)
 
+data Pattern = Pattern
+  { regex  :: Regex
+  , string :: String
+  }
+
+instance Show Pattern where show = show . string
+
+instance Eq Pattern where (==) = (==) `on` string
+
+instance IsString Pattern where
+  fromString = either error id . parsePattern
```

Comment:
The `error` seems a bit risky. Are we sure that's not going to sneak in somewhere unexpected? Would it be much worse to just have `unsafeParsePattern`?

Reply:
I tend to use partial functions in my own `IsString` instances because I figure that mistakes in string literals are very unlikely to make it past even cursory tests. Hasn't bitten me yet, but maybe I'm underestimating the problem :).

In this case, the instance is only used once and `Pattern` isn't exported from this module. That could change if we end up using `Pattern` in other parts of the code later though, so making it explicit makes sense. For now, I can inline the `either error id` bit into the one place where I used the `IsString` instance previously.

Reply:
Oh wait, never mind, I'll probably want to export the unsafe version for use in tests too.

Reply:
I've seen people use `fromString` directly, though. Not a huge deal.